src/Compression/Compression.cs

```diff
@@ -101,17 +101,17 @@ private void InitializeBuffer ()
 
 		private void EnsureBufferInitialized ()
```

Comment:
Please read https://docs.microsoft.com/en-us/dotnet/csharp/nullable-warnings#possible-dereference-of-null specially the section with the following method signature:
```csharp
private static bool IsNotNull([NotNullWhen(true)] object? obj) => obj != null;
```
We should follow the same pattern.